internal/interpreters/r_interpreter_functional_test.go

```diff
@@ -0,0 +1,213 @@
+package interpreters
+
+// Copyright (C) 2025 by Posit Software, PBC.
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/posit-dev/publisher/internal/logging"
+	"github.com/posit-dev/publisher/internal/types"
+	"github.com/posit-dev/publisher/internal/util"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/suite"
+)
+
+// RInterpreterFunctionalSuite contains functional tests that actually
+// interact with real R installations
+type RInterpreterFunctionalSuite struct {
+	suite.Suite
+	testProjectDir     util.AbsolutePath
+	log                logging.Logger
+	fs                 afero.Fs
+	interpreter        RInterpreter
+	rExecutable        util.AbsolutePath
+	defaultInterpreter *defaultRInterpreter
+	renvInstalled      bool
+}
+
+func TestRInterpreterFunctionalSuite(t *testing.T) {
+	suite.Run(t, new(RInterpreterFunctionalSuite))
+}
+
+func (s *RInterpreterFunctionalSuite) SetupTest() {
+	// Set up the test directory
+	parentDir, err := os.MkdirTemp("", "r-interpreter-test-*")
+	s.Require().NoError(err)
+
+	s.fs = afero.NewOsFs()
+	s.testProjectDir = util.NewAbsolutePath(parentDir, s.fs)
+	s.log = logging.New()
+
+	// Initialize R interpreter once for all tests
+	s.interpreter, err = NewRInterpreter(s.testProjectDir, util.Path{}, s.log, nil, nil, nil)
+	s.Require().NoError(err)
+
+	// Try to get R executable
+	s.rExecutable, err = s.interpreter.GetRExecutable()
+	if err != nil {
+		s.T().Logf("R executable not found: %v", err)
+		// Not failing here, individual tests will skip if needed
+	} else {
+		s.T().Logf("Using R executable: %s", s.rExecutable)
+		s.defaultInterpreter = s.interpreter.(*defaultRInterpreter)
+
+		// Check if renv is installed
+		aerr := s.defaultInterpreter.isRenvInstalled(s.rExecutable.String())
+		s.renvInstalled = (aerr == nil)
+		if !s.renvInstalled {
+			s.T().Logf("renv is not installed: %s", aerr.Error())
+		} else {
+			s.T().Logf("renv is installed and available")
+		}
+	}
+}
+
+// skipIfNoR skips the current test if R executable is not available
+// unless we're running in CI where we should enforce that R tests run
+func (s *RInterpreterFunctionalSuite) skipIfNoR() bool {
+	if os.Getenv("GITHUB_ACTIONS") != "true" && s.rExecutable.String() == "" {
+		s.T().Skip("Skipping test: R executable not found")
+		return true
+	}
+	return false
+}
```

Comment:
I would approach these as integration tests and run them outside of the unit tests. That way you don't have to conditionally check if R exists on the system.

Reply:
_nods_ let me see if I grok this before I implement it: if we took the `t.Short()` approach it would be adding something like 

```
   if testing.Short() {
        t.Skip("skipping functional tests in short mode.")
    }
```

And then in the Justfile add in `--short` by default (but in CI, when we want the functional tests, we don't use that). yeah?

Reply:
Yep, that's exactly how I was thinking about it.